### PR TITLE
Improve error message when loading non-PNG splash image

### LINKED ISSUE #96177: setup_boot_logo: Non-existing or invalid boot splash at 'res://icon.svg'. Loading default splash.
### Tested versions

Reproducible in v4.3.stable.official [77dcf97d8]

### System information

Debian Linux 11 x86_64

### Issue description

Unable to set any boot splash screen (even the included icon.svg)

### Steps to reproduce

Start Godot editor
Create new project
Project >> Project Settings
Enable: Advanced settings
Application >> Boot Splash
Image >> Open >> All Files (*)
Select: icon.svg
Image = res://icon.svg
Minimum Display Time = 3000
Close settings
Create root node: 2D Scene
Press F5 (run project)
Select current scene
Save when prompted
Splash screen does not show
Debugger error:
```
E 0:00:00:0460   setup_boot_logo: Non-existing or invalid boot splash at 'res://icon.svg'. Loading default splash.
  <C++ Source>   main/main.cpp:3185 @ setup_boot_logo()
```

### Minimal reproduction project (MRP)

[new-game-project.zip](https://github.com/user-attachments/files/16766144/new-game-project.zip)

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes https://github.com/godotengine/godot/issues/96177

Are there any other places that needs check?

### CODE REVIEW

File: editor/export/editor_export_platform.cpp [hunk 1]
```diff
@@ -881,7 +881,11 @@ Vector<String> EditorExportPlatform::get_forced_export_files() {
 		files.push_back(icon);
 	}
 	if (!splash.is_empty() && FileAccess::exists(splash) && icon != splash) {
-		files.push_back(splash);
+		if (!splash.ends_with(".png")) {
+			ERR_PRINT("Invalid boot splash : " + splash + ". The only supported format is PNG. Skipping export.");
+		} else {
+			files.push_back(splash);
+		}
```
Comment (timothyqiu) [inline editor/export/editor_export_platform.cpp:888]: I think the better error message in `setup_boot_logo()` is enough to solve the problem.

`get_forced_export_files()` does not seem to be a right place for this kind of checks.

Comment (jsjtxietian) [reply editor/export/editor_export_platform.cpp:888]: I was thinking about how we can prevent user doing this at the first place, I agree the export is not a right place lol.

**Review by timothyqiu — APPROVED** [review]

File: main/main.cpp [hunk 2]
```diff
@@ -3307,7 +3307,8 @@ void Main::setup_boot_logo() {
 				boot_logo.instantiate();
 				Error load_err = ImageLoader::load_image(boot_logo_path, boot_logo);
 				if (load_err) {
-					ERR_PRINT("Non-existing or invalid boot splash at '" + boot_logo_path + "'. Loading default splash.");
+					String msg = (boot_logo_path.ends_with(".png") ? "" : "The only supported format is PNG.");
+					ERR_PRINT("Non-existing or invalid boot splash at '" + boot_logo_path + +"'. " + msg + " Loading default splash.");
```
Comment (kleonc) [inline main/main.cpp:3311]: That's a potential double space here. :upside_down_face: One of the spaces should be a part of `msg`.

Comment (jsjtxietian) [reply main/main.cpp:3311]: And I noticed the double `+` too, lol

**Pull request merged**